### ISSUE TITLE
Mounting hole options

### ISSUE THREAD

**Issue by dtex** [makenai/sumobot-jr#16]
I'd like to add a parameter to the SCAD file for choosing different hole patterns for the top plate (Tessel 2 is my use case). Does that sound like a worthwhile PR?

**Comment by makenai** [comment]
Yup - most certainly. Thanks for looking at the SCAD files instead of the vectors. :)

I think you should be able to add an parameter / method to it similar to the arduino and pinoccio ones here: https://github.com/makenai/sumobot-jr/blob/master/3d_print/sumobot.scad#L186

**Comment by dtex** [comment]
Cool. I was thinking of adding a parameter like this:

```
// Mounting hole pattern
// valid values are "Arduino Uno", "Tessel 2"
hole_pattern = "Arduino Uno";
```

And then just add a conditional in `top` to call the right module.

**Comment by dtex** [comment]
Hmmm, like most things it's never as easy as it seems. I had to make the sled wider to fit the mounting holes, but the GPIO ports are still kind of exposed (anything plugged in could be easily ripped out by another sumo).

Sigh...

I'll keep playing with it.

![Sumobot with Tessel 2](https://lh3.googleusercontent.com/5xrnDFCnUzzvLkMw48IswVjbWoEvMtIYTicMNvIWVbrWeJMDmVOJY96Q0PapJ2ObTQWXwvt8QIvMFWiqwO19eCo-mbylxoey6vJMliKAqbk2-KbexF0sWR6TYyTMrPf-3LPmhGwICByweh_SR7cQNkGBYEOOZFRn3WsQR74C7cGE0u07YRoy4TRALWqKNiXEOgU_Llj1ADv6BLwTNiRBmN-VtjnghBIXdjJ1xK-RzTqJNnDgXXtvZk13vmUhOPw-u_6hYzUStgT8dNhXucd57UMxH3fJ7Oi4DE6GcYgij8FwGnz74Uz4OJwwXZGin0gSlJIlGOrO5KWQsd4L1KW11DKokuA8EBfR0R4GKlsTSR2dcHZbZIETlNlsPDbaX0OrXgfPtUOQuksn3AOjkafDGEu5526x9RQFTgV-Q_zlvgpWXlT0KZuKzp4sD6PMAQFTm-LRo8PEm0cT-CLIJgZKnecmOZgfDIAD_3RXnFZJUBX-0LWLMWP2fXix_n7GYOeS_uokdILAZBvH1vqVQuvhgL6cMmcw55oPQL5rVH9ElAhM7wyxyn-o88-JFNHoITeiwqiShlxkVkJeL7tjbqFoRM6dzPvxIa65=w1181-h886-no)

**Comment by dtex** [comment]
I'm thinking it makes the most sense to have an option for Tessel mounting holes like we discussed, but then just have a note somewhere (the readme?) that gives the minimum sled width and a suggested sled length... Well, maybe not. I don't see any reason we couldn't set the default width and height based on that same `hole_pattern` variable.
